### ISSUE TITLE
Python wheel package fails to build due to webpack dependency (jupyter)

### ISSUE THREAD

**Issue by elukey** [catboost/catboost#2530]
Problem: Building the Python wheel fails due to a webpack dependency on jupyter.
catboost version: HEAD
Operating System: Debian Bookworm
CPU: Intel i7

```
UserWarning: ValueError: Extensions require a devDependency on @jupyterlab/builder@^4.0.7, you have a dependency on 3.0.6
  warnings.warn(msg[-1].strip())
```

The following diff fixes the problem:

```
--- a/catboost/python-package/catboost/widget/js/package.json
+++ b/catboost/python-package/catboost/widget/js/package.json
@@ -21,7 +21,7 @@
     "plotly.js-dist-min": "^1.58.4"
   },
   "devDependencies": {
-    "@jupyterlab/builder": "^3.0.6",
+    "@jupyterlab/builder": "^4.0.7",
     "css-loader": "^5.2.0",
     "rimraf": "^3.0.2",
     "style-loader": "^2.0.0",
```

**Comment by andrey-khropov** [comment]
> The following diff fixes the problem:
> 
> ```
> --- a/catboost/python-package/catboost/widget/js/package.json
> +++ b/catboost/python-package/catboost/widget/js/package.json
> @@ -21,7 +21,7 @@
>      "plotly.js-dist-min": "^1.58.4"
>    },
>    "devDependencies": {
> -    "@jupyterlab/builder": "^3.0.6",
> +    "@jupyterlab/builder": "^4.0.7",
>      "css-loader": "^5.2.0",
>      "rimraf": "^3.0.2",
>      "style-loader": "^2.0.0",
> ```

Unfortunately it is not as simple as that because it switches the build-time dependency from jupyterlab 3.x to jupyterlab 4.x and it can be problematic for compatibility because [jupyterlab 4.x introduced changes that are not backwards - compatible with jupyterlab 3.x](https://jupyterlab.readthedocs.io/en/latest/extension/extension_migration.html#jupyterlab-3-x-to-4-x):

* CatBoost widget has not been checked under jupyterlab 4.x. I've created #2533 for that.
* CatBoost widget built with jupyterlab/builder 4.x will possibly won't work with jupyterlab 3.x (and jupyterlab 3.x is still supported and widely used so we shouldn't drop its support immediately). This has to be checked as well and widget code adapted to be compatible with both jupyterlab 3.x and jupyterlab 4.x and possibly two versions of the widget to be released in the future.

In commit https://github.com/catboost/catboost/commit/76ed5da2bc96bd28e6fc0c76527cd94c970c9562 I've changed `jupyterlab` version specification in dependencies to make clear that only `jupyterlab` 3.x versions are supported for now.
